src/human_player_module.c

```diff
@@ -0,0 +1,391 @@
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include "cards_management.h"
+#include "human_player_module.h"
+
+/* Global variables */
+bool g_card_requested = false;
+bool is_action_card = false;
+Player_t *human_player;
+
+int request_card(PlayerType_e PlayerType, int no_of_cards);
+void invalid_turn_warning();
+const Deck_t *show_recent_card(Deck_t **pp_head);
+PlayerType_e determine_next_player(struct CARD *previous_card, PlayerType_e current_player);
+bool validate_card(char *entered_value);
+Card_t *map_user_input(char *input);
+Card_t *pick_card_from_deck(Deck_t **pp_head, Card_t *card_to_be_matched); //Can be included in Card management
+error_e remove_card_from_deck(Deck_t **pp_head, Card_t *card_to_be_removed); //Can be included in Card management
+error_e record_human_input();
+void show_cards_assigned(Card_t assigned_card); 
+bool check_is_valid_turn(Card_t *previous_card, Card_t *current_card);
+void display_player_deck(Player_t *player);
+void display_player_turn(PlayerType_e next_player);
+void end_turn(PlayerType_e type);
+int quit_game();
+error_e handle_human_turn();
+
+//Write for computer as well?
+/**
+ * @brief Function called when the human player requests a card during each turn/
+ *  Draw1/ Draw2 
+ * Assigns the top most card/cards in the draw pile to the user
+ *
+ * @param PlayerType - Indicates whether user requested is Human or Computer Player
+ * @param no_of_cards - Indicates the number of cards requested  
+ */
+int request_card(PlayerType_e PlayerType, int no_of_cards)
```

Comment:
Since the function is belong to human_player_module.c now, let's remove the input parameter: PlayerType_e PlayerType. If we need it for computer as well, then we will refractor the function and move it to cards_management.c.